codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSHttpProtocolServiceClient.kt

```diff
@@ -156,7 +157,14 @@ class AWSHttpProtocolServiceClient(
                             writer.write("region,")
                         }
                         "awsCredentialIdentityResolver" -> {
-                            writer.write("\$N(),", AWSSDKIdentityTypes.DefaultAWSCredentialIdentityResolverChain)
+                            if (ctx.settings.visibility == "package") {
+                                writer.write(
+                                    "\$N(),",
+                                    InternalAWSCommonTypes.EmptyAWSCredentialIdentityResolver,
+                                )
+                            } else {
+                                writer.write("\$N(),", AWSSDKIdentityTypes.DefaultAWSCredentialIdentityResolverChain)
+                            }
```

Comment:
Here & below, default value for awsCredentialIdentityResolver is set differently for internal or normal clients.